plugins/module_utils/template.py

```diff
@@ -566,3 +566,59 @@ def get_fabric_template_object_by_key_value(self, object_type, object_descriptio
         match = self.get_object_by_key_value_pairs(object_description, search_list, kv_list, fail_module)
         if match:
             return match.details
+
+    def get_fabric_span_session(self, uuid=None, name=None, fail_module=False):
+        """
+        Get the Fabric SPAN Session by uuid or name.
+        :param uuid: UUID of the Fabric SPAN Session to search for -> Str
+        :param name: Name of the Fabric SPAN Session to search for -> Str
+        :param fail_module: When match is not found fail the ansible module -> Bool
+        :return: Dict | None | List[Dict] | List[]: The processed result which could be:
+                 When the UUID | Name is existing in the search list -> Dict
+                 When the UUID | Name is not existing in the search list -> None
+                 When both UUID and Name are None, and the search list is not empty -> List[Dict]
+                 When both UUID and Name are None, and the search list is empty -> List[]
+        """
+        existing_objects = self.template.get("monitoringTemplate", {}).get("template", {}).get("spanSessions", [])
+        if uuid or name:  # Query a specific object
+            return self.get_object_by_key_value_pairs("SPAN Sessions", existing_objects, [KVPair("uuid", uuid) if uuid else KVPair("name", name)], fail_module)
```

Comment:
is plural form correct here?
```suggestion
            return self.get_object_by_key_value_pairs("SPAN Session", existing_objects, [KVPair("uuid", uuid) if uuid else KVPair("name", name)], fail_module)
```

Reply:
applies to all 3